### PR TITLE
test: repro #7430

### LINKED ISSUE #7430: `dune describe pp` creates an empty `profile.dump` file at the root of the project
## Expected Behavior

`dune describe pp some/file.ml` should print the preprocess file.

## Actual Behavior

It does print the preprocessed file but also creates a `profile.dump` file at the root of the project. This file is empty. I couldn't find a mention of this file in `_build/log`,

## Reproduction

```shell
git clone https://github.com/hhugo/merge-fmt
dune describe pp test/merge.ml
```

## Specifications

- Version of `dune` (output of `dune --version`): 3.6.1 (but also happening on an experimental version of dune)
- Version of `ocaml` (output of `ocamlc --version`): 4.14.0
- Operating system (distribution and version): debian 11

## Additional information 

- Link to gist with verbose output (run `dune` with the `--verbose` flag): https://gist.github.com/Khady/9660bc040c9c8fc3af90aa50e1fab0cc

### PR DESCRIPTION
**Pull request by Alizter**
Test case for #7430. We demonstrate that the `dune describe pp` command
leaves behind the dump file.

### CODE REVIEW

**Comment by emillon** [comment]
The PR title says it's a test repro, but it also contains a fix. Can you put the test in a separate PR?

**Comment by Alizter** [comment]
@emillon Done.

File: test/blackbox-tests/test-cases/describe/describe-pp/describe-pp.t/run.t [hunk 1]
```diff
@@ -7,3 +7,12 @@ We can also show the original source if it is not preprocessed
 
   $ dune describe pp src/util.ml
   let log str = print_endline str
+
+We also make sure that the dump file is not present
+
+  $ ls
```
Comment (emillon) [inline test/blackbox-tests/test-cases/describe/describe-pp/describe-pp.t/run.t:13]:
```suggestion
  $ dune_cmd exists profile.dump
```

I'm not sure the output of `ls` is stable enough for this

Comment (Alizter) [reply test/blackbox-tests/test-cases/describe/describe-pp/describe-pp.t/run.t]: done.

**Review by emillon — APPROVED** [review]

**Pull request merged**